src/TSTransformer/nodes/transformSourceFile.ts

```diff
@@ -205,13 +205,26 @@ export function transformSourceFile(state: TransformState, node: ts.SourceFile)
 		}
 	}
 
+	const headerStatements = luau.list.make<luau.Statement>();
+
+	// add build information to the tree
+	luau.list.push(headerStatements, luau.comment(` Compiled with roblox-ts v${COMPILER_VERSION}`));
+
 	// add the Runtime library to the tree if it is used
 	if (state.usesRuntimeLib) {
-		luau.list.unshift(statements, state.createRuntimeLibImport(node));
+		luau.list.push(headerStatements, state.createRuntimeLibImport(node));
 	}
 
-	// add build information to the tree
-	luau.list.unshift(statements, luau.comment(`Compiled with roblox-ts v${COMPILER_VERSION}`));
+	// extract Luau directive comments like --!strict so we can put them before headerStatements
+	const directiveComments = luau.list.make<luau.Statement>();
+	while (statements.head && luau.isComment(statements.head.value) && statements.head.value.text.startsWith("!")) {
+		const comment = luau.list.shift(statements);
+		assert(comment, "Expected comment statement!");
+		luau.list.push(directiveComments, comment);
+	}
+
+	luau.list.unshiftList(statements, headerStatements);
+	luau.list.unshiftList(statements, directiveComments);
```

Comment:
While not a big deal, this extra list could be optimized away by swapping the execution order, pushing the directives onto the empty headerStatements first and only then doing the RuntimeLib logic

Reply:
I considered this, but I think the performance difference is negligible and this reads easier at first glance.